message.go

```diff
@@ -57,6 +60,9 @@ func (m *Message) encode(pe packetEncoder) error {
 	pe.putInt8(m.Version)
 
 	attributes := int8(m.Codec) & compressionCodecMask
+	if m.LogAppendTime {
+		attributes |= timestampTypeMask
```

Comment:
by @mikejquinn:
> Does this make sense to be set by a producer? I thought this was a server config setting.

This change is done for the sarama tests to pass. The tests use `encode`/`decode` to simulate message delivery. So, yes, normally the flag is only set by the Kafka and should not be considered there.

I think I could change the tests to overcome this, but I am not sure it worth the efforts.